doc: Add useful V8 option section (#32262)
nodejs/node: doc/api/cli.md

greater than `4` (its current default value). For more information, see the
[libuv threadpool documentation][].

> ## Useful v8 options

Reviewer: ```suggestion
## Useful V8 options
```

> These are v8 options and fall outside Node.js' responsibility.

Reviewer: ```suggestion
These are V8 options and fall outside Node.js' responsibility.
```

> These are v8 options and fall outside Node.js' responsibility.

Reviewer 2: ```suggestion
V8 has CLI options. Any V8 CLI option that is provided to `node` will be passed on to V8 to handle. V8's options have _no stability guarantee_. The V8 team themselves don't consider them to be part of their formal API, and reserve the right to change them at any time. The Node.js team does not consider them covered by the Node.js stability guarantees. Many of the V8 options are of interest only to V8 developers. Despite this, there is a small set of V8 options that are widely applicable to Node.js, and they are documented here.
```

I think we should say something like the above.

> V8 options that are allowed are:

Reviewer: I think this should either be removed completely or at least reworded. Node does not restrict the V8 options that can be passed on the command line.

Author: There is another place in the document mentioning the same thing.

> ### `--max-old-space-size`

Reviewer 2: ```suggestion
### `--max-old-space-size=SIZE` (in Mbytes) 
```

Sets the max memory size of V8 old memory section. As memory consumption approaches the limit, V8 will spend more time on garbage collection in an effort to free unused memory.

> On a machine with 2GB of memory it suggested to set 1.5GB to leave some memory for other uses and avoid swapping.

Reviewer 3: ```suggestion
Sets the max memory size of V8 old memory section. As memory
consumption approaches the limit, V8 will spend more time on
garbage collection in an effort to free unused memory.

On a machine with 2GB of memory it suggested to set 1.5GB to
leave some memory for other uses and avoid swapping.
```

nit: need to wrap lines at <= 80 chars

greater than `4` (its current default value). For more information, see the
[libuv threadpool documentation][].

## Useful V8 options

V8 has its own set of CLI options. Any V8 CLI option that is provided to `node` will
be passed on to V8 to handle. V8's options have _no stability guarantee_.
The V8 team themselves don't consider them to be part of their formal API,
and reserve the right to change them at any time. The Node.js team does not

> consider them covered by the Node.js stability guarantees. Many of the V8

Reviewer 4: Optional nit:

```suggestion
and reserve the right to change them at any time. Likewise, they are not
covered by the Node.js stability guarantees. Many of the V8
```

options are of interest only to V8 developers. Despite this, there is a small

> set of V8 options that are widely applicable to the Node.js, and they are

Reviewer: ```suggestion
set of V8 options that are widely applicable to Node.js, and they are
```

> documented here. V8 options that are allowed are:

Reviewer: ```suggestion
documented here:
```

### `--max-old-space-size=SIZE` (in Mbytes)

> Sets the max memory size of V8 old memory section. As memory

Reviewer: ```suggestion
Sets the max memory size of V8's old memory section. As memory
```

consumption approaches the limit, V8 will spend more time on
garbage collection in an effort to free unused memory.

> On a machine with 2GB of memory it is suggested to set this to

Reviewer 4: Missing comma:

```suggestion
On a machine with 2GB of memory, it is suggested to set this to
```

I'm not a fan of "it is suggested" or "it is recommended" formulations. Either tell people to do something or don't.  If you can't make a blanket statement, at least avoid the passive voice with something like "consider":

```suggestion
On a machine with 2GB of memory, consider setting this to
```

Author: Understood.

> On a machine with 2GB of memory it suggested to set 1.5GB

Reviewer: ```suggestion
On a machine with 2GB of memory it is suggested to set this to 1500
```

1536 (1.5GB) to leave some memory for other uses and avoid swapping.

> E.g. `node --max-old-space-size=1536 index.js`

Reviewer 4: Optional nit: Delete this line.

Author: I personally feel an example helps a lot when introducing to a flag. It clearly presents how to pass an argument.

Reviewer 4: > I personally feel an example helps a lot when being introduced to a flag

If we want to provide an example, it should conform with the formatting of the other examples provided elsewhere in the doc. Otherwise, each entry is inventing its own example formatting, and we definitely do not want that. The existing format is more useful than this as it provides context, such as what the anticipated output might look like. So, this example might show the error one might get if running without the flag, then show the command succeeding with the flag.

(That said, most of our entries do not contain examples and that seems appropriate in this document.)

Author: Totally agree with your point, Can you point me to some of the correct existing formats?
What I see a mixture of formats, some given below
1. node --name-of-flag index.mjs
2. node --name-of-flag <file>

Reviewer 4: I don't mean the format of the command itself. I mean the markdown formatting that denotes an example.

````markdown
```console
$ node --cpu-prof index.js
$ ls *.cpuprofile
CPU.20190409.202950.15293.0.0.cpuprofile
```
````

Author: @Trott checkout the formatting once now.
